platforms/posix/src/px4/common/main.cpp

```diff
@@ -286,6 +286,10 @@ int main(int argc, char **argv)
 			commands_file = "etc/init.d-posix/rcS";
 		}
 
+		if (working_directory.empty()){
```

Comment:
```suggestion
		if (working_directory.empty()) {
```

Reply:
Check if the path exists.

Reply:
Alright, I decided to only set a default working directory with the `PX4_SOURCE_DIR` only if `PX4_SOURCE_DIR` is defined, and also added checking of the `working_directory` path